### PR TITLE
Automatically determine ssl parameter when uri has https scheme (#134)

### PR DESCRIPTION
**Pull request by reify-james-vickers**
https://github.com/protojure/lib/issues/133

Convenience enhancement so caller doesn't have to specify `:uri` for common situations.  If uri starts with `https://` scheme, then use ssl as true. Still respect provided `:ssl` param if present.

Added tests for this situation that use circleci.bond for this purpose, since it's difficult to setup an SSL test; this verifies the expected connect call is made.

### CODE REVIEW

File: modules/grpc-client/src/protojure/grpc/client/providers/http2.clj [hunk 1]
```diff
@@ -34,8 +34,8 @@ A map with the following entries:
 A promise that, on success, evaluates to an instance of [[api/Provider]].
 _(api/disconnect)_ should be used to release any resources when the connection is no longer required.
   "
-  [{:keys [uri codecs content-coding max-frame-size input-buffer-size metadata idle-timeout ssl] :or {codecs builtin-codecs max-frame-size 16384 input-buffer-size jetty/default-input-buffer ssl false} :as params}]
+  [{:keys [uri codecs content-coding max-frame-size input-buffer-size metadata idle-timeout ssl] :or {codecs builtin-codecs max-frame-size 16384 input-buffer-size jetty/default-input-buffer} :as params}]
   (log/debug "Connecting with GRPC-HTTP2:" params)
   (let [{:keys [host port]} (lambdaisland/uri uri)]
-    (-> (jetty/connect {:host host :port (Integer/parseInt port) :input-buffer-size input-buffer-size :idle-timeout idle-timeout :ssl ssl})
+    (-> (jetty/connect {:host host :port (Integer/parseInt port) :input-buffer-size input-buffer-size :idle-timeout idle-timeout :ssl (or ssl (.startsWith (.toLowerCase uri) "https://"))})
```
Comment (ghaskins) [inline modules/grpc-client/src/protojure/grpc/client/providers/http2.clj:40]: minor nit: why not use clojure.string [starts-with? lower-case] ?

**Review by ghaskins — CHANGES_REQUESTED** [review]
One minor comment inline.  Also, feel free to add yourself to CONTRIBUTORS.md.  Thank you for the patch.

File: modules/grpc-client/src/protojure/grpc/client/providers/http2.clj (hunk 1 above)
Comment (reify-james-vickers) [reply modules/grpc-client/src/protojure/grpc/client/providers/http2.clj]: Good idea, did that in f08a303dd2d38ef542e84fcfb1c66c1e3907f44e

**Review by ghaskins — APPROVED** [review]
Thank you!

**Pull request merged**